src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -35,6 +36,108 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
             return result;
         }
 
+        private BoundNode RewriteToInterpolatedStringBuilderPattern(BoundInterpolatedString node)
+        {
+            Debug.Assert(node.InterpolationData is { Construction: not null });
+            // PROTOTYPE(interp-string): Support the general builder pattern that doesn't save to a local.
+            var resultSymbol = _factory.SynthesizedLocal(_compilation.GetSpecialType(SpecialType.System_String), node.Syntax);
+            var resultTemp = _factory.Local(resultSymbol);
+            var builderTempSymbol = _factory.InterpolatedStringBuilderLocal(node.InterpolationData.BuilderType, node.InterpolationData.ScopeOfContainingExpression, node.Syntax);
+            var builderTemp = _factory.Local(builderTempSymbol);
+
+            // PROTOTYPE(interp-string): Support optional out param for whether the builder was created successfully
+            // var builder = Construction(baseStringLength, numFormatHoles);
+            var builderConstruction = _factory.Assignment(builderTemp, MakeCallWithNoExplicitArgument(node.InterpolationData.Construction, node.Syntax, expression: null, assertParametersAreOptional: false));
+
+            // For every interpolation part, call the appropriate builder.TryFormat... method.
+
+            Debug.Assert(node.InterpolationData.BuilderFormatCalls.Length >= 1);
+            Debug.Assert(node.Parts.Length == node.InterpolationData.BuilderFormatCalls.Length);
+            var usesBoolReturn = node.InterpolationData.BuilderFormatCalls[0].Method.ReturnType.SpecialType == SpecialType.System_Boolean;
+            var tryFormatCalls = ArrayBuilder<BoundExpression>.GetInstance(node.InterpolationData.BuilderFormatCalls.Length);
+            for (int i = 0; i < node.Parts.Length; i++)
+            {
+                var formatInfo = node.InterpolationData.BuilderFormatCalls[i];
+                var part = node.Parts[i];
+                Debug.Assert(usesBoolReturn == (formatInfo.Method.ReturnType.SpecialType == SpecialType.System_Boolean));
+                Debug.Assert(formatInfo.Arguments[0] is BoundInterpolatedStringElementPlaceholder { Index: var index } && index == i);
+                Debug.Assert(part is BoundStringInsert or BoundLiteral);
+
+                // Visit the part, and replace the part placeholder in the TryFormat... call with the replaced part.
+                var visitedPart = VisitExpression(part is BoundStringInsert { Value: var value } ? value : part);
+                var formatCall = MakeCallWithNoExplicitArgument(formatInfo with
```

Comment:
>MakeCallWithNoExplicitArgument [](start = 33, length = 30)

The use of this method doesn't feel appropriate because it is not meant to be used with arguments. Even if it appears to be working, it is not meant to deal with arguments produced by initial binding.